### PR TITLE
Convert to PDK and Puppet 7

### PR DESCRIPTION
**Pull request by Phil-Friderici**
This collection of commits does convert the module to Puppet 7 and PDK compatibility. It satisfies puppet-lint and robucops.

I would like to get this PR reviewed and merged before continueing to work on the module and unit tests.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,290 +1,359 @@
...
+#   File with the Postfix SMTP server RSA private key in PEM format.
+#
+# @param main_smtpd_tls_mandatory_protocols
+#  List of SSL/TLS protocols that the Postfix SMTP server will use with mandatory
+#  TLS encryption. An empty value means allow all protocols. The valid protocol
+#  names, are "SSLv2", "SSLv3" and "TLSv1". The default value is "!SSLv2, !SSLv3"
+#  for Postfix releases after the middle of 2015, "!SSLv2" for older releases.
+#
+# @param main_smtpd_tls_protocols
+#   FIXME: Add description
```
Comment (anders-larsson) [inline manifests/init.pp]: Description? ;)

**Review by anders-larsson — COMMENTED** [review]
Looks very good. Only thing I can find is the FIXME entry in header which should probably be fixed.

Love the simplification allowed by using modul hiera.

File: manifests/init.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/init.pp]: added the missing description

**Comment by Phil-Friderici** [comment]
Thanks for your review, I guess now this PR is complete :)

**Comment by anders-larsson** [comment]
BTW. Would it make more sense to add the "current" default to say Ubuntu.yaml in Hiera and then in cases where it is something different it would be added to a Ubuntu-14.04 YAML file? This would make the module automatically support new releases (unless they're different)?

Or do you think it is better to be explicit?

So in the case of Ubuntu it would be:
Ubuntu.yaml
```
postfix::main_daemon_directory: '/usr/lib/postfix/sbin'
```
Ubuntu-14.04.yaml
```
 postfix::main_daemon_directory: '/usr/lib/postfix' 
```

**Comment by Phil-Friderici** [comment]
That's a good question, I am not sure myself what I would prefer. I like to be explicit because it eleminates guessworking. On the other side it brings the 'cost' of adding support for new versions later on.

In the SSH module (v4 branch) I have created explicit configuration files for each OS family and variation. I have to confess that I like it a lot. All settings for an specific OS are collected in hiera files :)

But please let us continue that question outside of this PR that is dedicated to migrate to Puppet 7. I would like to create new PRs for refactoring the unit tests to start using rspec-puppet-facts and anothers for refactoring details like the above.

**Comment by anders-larsson** [comment]
Sure. Let's get this merged!

**Pull request merged**